pkgs/development/node-packages/node-packages.nix

```diff
@@ -120864,6 +120981,39 @@ in
     bypassCache = true;
     reconstructLock = true;
   };
+  ts-node = nodeEnv.buildNodePackage {
+    name = "ts-node";
+    packageName = "ts-node";
+    version = "10.4.0";
+    src = fetchurl {
+      url = "https://registry.npmjs.org/ts-node/-/ts-node-10.4.0.tgz";
+      sha512 = "g0FlPvvCXSIO1JDF6S232P5jPYqBkRL9qly81ZgAOSU7rwI0stphCgd2kLiCrU9DjQCrJMWEqcNSjQL02s6d8A==";
+    };
+    dependencies = [
+      sources."@cspotcode/source-map-consumer-0.8.0"
+      sources."@cspotcode/source-map-support-0.7.0"
+      sources."@tsconfig/node10-1.0.8"
+      sources."@tsconfig/node12-1.0.9"
+      sources."@tsconfig/node14-1.0.1"
+      sources."@tsconfig/node16-1.0.2"
+      sources."acorn-8.5.0"
+      sources."acorn-walk-8.2.0"
+      sources."arg-4.1.3"
+      sources."create-require-1.1.1"
+      sources."diff-4.0.2"
+      sources."make-error-1.3.6"
+      sources."yn-3.1.1"
```

Comment:
you will need to add the peerdependencies to that list https://github.com/nocodb/nocodb/blob/master/packages/nocodb/Dockerfile#L28
typescript is defined as a peerdependency and not as a normal dependency.
I'm not sure if there is a setting to include peerdependencies in node2nix.